olm/src/jsMain/kotlin/io/github/matrixkt/olm/JsOlm.kt

```diff
@@ -0,0 +1,160 @@
+        public fun pickle(key: String): String;
+        public fun pickle(key: Uint8Array): String;
+        public fun create(session_key: String): String;
+        public fun session_id(): String;
+        public fun first_known_index(): Long;
+        public fun free();
+        public fun import_session(session_key: String): String;
+
+        public fun export_session(message_index: Long): String;
+        public fun decrypt(message: String): DecryptedMessage;
+        public fun unpickle(key: Uint8Array, pickle: String);
+        public fun unpickle(key: String, pickle: String);
+    }
+
+    public class OutboundGroupSession {
+        public constructor();
+        public fun free();
+        public fun pickle(key: Uint8Array): String;
+        public fun pickle(key: String): String;
+        public fun unpickle(key: Uint8Array, pickle: String);
+        public fun unpickle(key: String, pickle: String);
+        public fun create();
+        public fun encrypt(plaintext: String): String;
+        public fun session_id(): String;
+        public fun session_key(): String;
+        public fun message_index(): Int;
+    }
+
+    public class PkDecryption {
+        public fun constructor();
+        public fun free();
+        public fun init_with_private_key(key: Uint8Array): String;
+        public fun generate_key(): String;
+        public fun get_private_key(): Uint8Array;
+        public fun pickle(key: Uint8Array): String;
+        public fun pickle(key: String): String;
+        public fun unpickle(key: Uint8Array, pickle: String): String;
+        public fun unpickle(key: String, pickle: String): String;
+        public fun decrypt(ephemeral_key: String, mac: String, ciphertext: String): String;
+    }
+
+    public class PkEncryption {
+        public fun constructor();
+        public fun free();
+        public fun set_recipient_key(key: String);
+        public fun encrypt(plaintext: String): OlmPkMessage;
+    }
+
+    public class SAS {
+        public fun free();
+        public fun get_pubkey(): String;
+        public fun set_their_key(their_key: String);
+        public fun generate_bytes(info: String, length: Int): Uint8Array;
+        public fun calculate_mac(input: String, info: String): String;
+        public fun calculate_mac_long_kdf(input: String, info: String): String;
+        public fun is_their_key_set(): Boolean;
+    }
+
+    public class Utility {
+        public constructor();
+        public fun free();
+        public fun sha256(input: String): String;
+        public fun sha256(input: Uint8Array): String;
+        public fun ed25519_verify(key: String, message: String, signature: String);
+        public fun ed25519_verify(key: String, message: Uint8Array, signature: String);
+    }
+
+}
+
+public fun Uint8Array.toByteArray() : ByteArray {
+    var result = ByteArray(this.byteLength)
+    for (i in 0..this.byteLength) {
+        result[i] = this[i]
+    }
+    return result
```

Comment:
```suggestion
    return ByteArray(this.byteLength) { this[it] }
```

Reply:
Thanks i googled a lot but could not find a nice way to do it. this is way cleaner.